### ISSUE TITLE
pom.xml is flagging a number of issues (a FATAL error & almost 13 warnings) post importing 'serenity-rest-starter' 

### ISSUE THREAD

**Issue by TarahMo** [serenity-bdd/serenity-rest-starter#7]
Hi John

I have tried importing the 'serenity-rest-starter' (cloning git & importing as existing maven project) in Eclipse IDE.

However, the pom.xml is flagging a number of issues (a FATAL error & almost 13 warnings).

I have also noticed that it is not creating Maven Dependencies folder within the framework (Tried updating the Maven project as well as the Force Update)

Please can you advise as to why I would have these issues in pom.xml (I just have imported the maven project and have done no customizations/enhancements to the project) ? Attached a few error screen captures in addition to the details below.

Error (FATAL):

1 problem was encountered while building the effective model
[FATAL] Non-readable POM C:\Users\Username\.m2\repository\org\springframework\boot\spring-boot-maven-plugin\2.4.2\spring-boot-maven-plugin-2.4.2.pom: input contained no data @ 

Warnings:

Warning 1: Overriding managed version 1.2.3 for logback-classic

Warning 2: Overriding managed version 3.3.0 for rest-assured

Warning 3: Plugin could not be resolved. Ensure the plugin's groupId, artifactId and version are present.

Additional information: Plugin org.springframework.boot:spring-boot-maven-plugin:2.4.2 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.springframework.boot:spring-boot-maven-plugin:jar:2.4.2

Warning 4: Overriding managed version 2.5.14 for groovy

Warning 5: Overriding managed version 2.5.14 for groovy-xml

Warning 6: Overriding managed version 3.3.0 for json-path

Warning 7: Overriding managed version 3.3.0 for xml-path

Warning 8: Overriding managed version 2.22.2 for maven-surefire-plugin

Warning 9: Overriding managed version 2.22.2 for maven-failsafe-plugin

Warning 10: Duplicating managed version 4.13.1 for junit

Warning 11: Overriding managed version 3.18.1 for assertj-core

Warning 12: Overriding managed version 3.8.1 for maven-compiler-plugin

Warning 13: Duplicating managed version 2.4.2 for spring-boot-maven-plugin

![Capture 1](https://user-images.githubusercontent.com/85223433/120466241-750d3580-c3bc-11eb-9215-612d85fac48b.JPG)

![Capture 2](https://user-images.githubusercontent.com/85223433/120466283-81918e00-c3bc-11eb-89e3-3f1972f175bb.JPG)

![Capture 3](https://user-images.githubusercontent.com/85223433/120466312-89513280-c3bc-11eb-9127-7fcc08f8d889.JPG)

**Comment by wakaleo** [comment]
That's an Eclipse problem of some kind. I don't know what the work-around is - maybe ask the Eclipse community?

**Comment by TarahMo** [comment]
I shall try reaching them but in the meantime are you able to suggest a possible work-around from your view of these issues ? probably uninstall-and install the IDE to see if that offers a better situation ? Are all of these (including the 14 warnings) related to Eclipse in your view ?

**Comment by wakaleo** [comment]
Have you considered using IntelliJ?

**Comment by TarahMo** [comment]
Intellij did help in resolving the errors reported above so certainly they are all eclipse ! - thank you for suggesting and I must say its a much better IDE. 

1) I then tried importing 'serenity-rest-starter-master' into Intellij but encountered the error below. 

java.lang.IllegalAccessError: class lombok.javac.apt.LombokProcessor cannot access class com.sun.tools.javac.processing.JavacProcessingEnvironment [duplicate]

I did some troubleshooting and noticed this can be resolved by adding the below dependency to pom.xml - prefect ! - that resolved the error and the build ran successfully (Maven -> Reload Project & Build Project)

![Capture GCX](https://user-images.githubusercontent.com/85223433/120551274-74ec5480-c413-11eb-800f-83d620880b03.JPG)

2) Before doing customisations (adding features files, step definitions etc.) for the 'Best Buy API Playground' API hosted on my localhost, I wanted to ensure the 'serenity-rest-starter-master' git project runs with no errors, thus I ran 'CucumberTestSuite' (without doing any modifications to the git starter project) - However, it reported that 3 test failed and the following problems have been reported.

Please can you advise re these errors while running the git starter project ? I did troubleshoot but couldn't find anything significant that can fix these errors.

net.serenitybdd.core.exceptions.SerenityManagedException
	at net.serenitybdd.rest.utils.RestDecorationHelper.decorate(RestDecorationHelper.java:20)
	at net.serenitybdd.rest.SerenityRest.fetchRequestSpecification(SerenityRest.java:233)
	at net.serenitybdd.rest.SerenityRest.given(SerenityRest.java:220)
	at starter.postcodes.PostCodeAPI.fetchLocationByPostCodeAndCountry(PostCodeAPI.java:13)
	at starter.postcodes.PostCodeAPI$ByteBuddy$k8KZor31.fetchLocationByPostCodeAndCountry$accessor$keU7q0qG(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.thucydides.core.steps.StepInterceptor.invokeMethod(StepInterceptor.java:492)
	at net.thucydides.core.steps.StepInterceptor.executeTestStepMethod(StepInterceptor.java:472)
	at net.thucydides.core.steps.StepInterceptor.runTestStep(StepInterceptor.java:445)
	at net.thucydides.core.steps.StepInterceptor.runOrSkipMethod(StepInterceptor.java:201)
	at net.thucydides.core.steps.StepInterceptor.testStepResult(StepInterceptor.java:188)
	at net.thucydides.core.steps.StepInterceptor.intercept(StepInterceptor.java:76)
	at net.thucydides.core.steps.ProxyConfiguration$InterceptorDispatcher.intercept(ProxyConfiguration.java:51)
	at starter.postcodes.PostCodeAPI$ByteBuddy$k8KZor31.fetchLocationByPostCodeAndCountry(Unknown Source)
	at starter.postcodes.PostCodeStepDefinitions.lookUpAPostCode(PostCodeStepDefinitions.java:19)
	at ✽.I look up a post code 10000 for country code US(classpath:features/postcodes/looking_up_post_codes.feature:4)
Caused by: java.lang.UnsupportedOperationException: Cannot define class using reflection: Unsupported class file major version 60
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection$Dispatcher$Initializable$Unavailable.defineClass(ClassInjector.java:409)
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection.injectRaw(ClassInjector.java:234)
	at net.bytebuddy.dynamic.loading.ClassInjector$AbstractBase.inject(ClassInjector.java:110)
	at net.bytebuddy.dynamic.loading.ClassLoadingStrategy$Default$InjectionDispatcher.load(ClassLoadingStrategy.java:233)
	at net.bytebuddy.dynamic.loading.ClassLoadingStrategy$Default.load(ClassLoadingStrategy.java:144)
	at net.bytebuddy.dynamic.TypeResolutionStrategy$Passive.initialize(TypeResolutionStrategy.java:100)
	at net.bytebuddy.dynamic.DynamicType$Default$Unloaded.load(DynamicType.java:6156)
	at net.serenitybdd.rest.utils.RestSpecificationFactory.<clinit>(RestSpecificationFactory.java:57)
	at net.serenitybdd.rest.utils.RestDecorationHelper.decorate(RestDecorationHelper.java:20)
	at net.serenitybdd.rest.SerenityRest.fetchRequestSpecification(SerenityRest.java:233)
	at net.serenitybdd.rest.SerenityRest.given(SerenityRest.java:220)
	at starter.postcodes.PostCodeAPI.fetchLocationByPostCodeAndCountry(PostCodeAPI.java:13)
	at starter.postcodes.PostCodeAPI$ByteBuddy$k8KZor31.fetchLocationByPostCodeAndCountry$accessor$keU7q0qG(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.thucydides.core.steps.StepInterceptor.invokeMethod(StepInterceptor.java:492)
	at net.thucydides.core.steps.StepInterceptor.executeTestStepMethod(StepInterceptor.java:472)
	at net.thucydides.core.steps.StepInterceptor.runTestStep(StepInterceptor.java:445)
	at net.thucydides.core.steps.StepInterceptor.runOrSkipMethod(StepInterceptor.java:201)
	at net.thucydides.core.steps.StepInterceptor.testStepResult(StepInterceptor.java:188)
	at net.thucydides.core.steps.StepInterceptor.intercept(StepInterceptor.java:76)
	at net.thucydides.core.steps.ProxyConfiguration$InterceptorDispatcher.intercept(ProxyConfiguration.java:51)
	at starter.postcodes.PostCodeAPI$ByteBuddy$k8KZor31.fetchLocationByPostCodeAndCountry(Unknown Source)
	at starter.postcodes.PostCodeStepDefinitions.lookUpAPostCode(PostCodeStepDefinitions.java:19)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at io.cucumber.java.Invoker.doInvoke(Invoker.java:66)
	at io.cucumber.java.Invoker.invoke(Invoker.java:24)
	at io.cucumber.java.AbstractGlueDefinition.invokeMethod(AbstractGlueDefinition.java:47)
	at io.cucumber.java.JavaStepDefinition.execute(JavaStepDefinition.java:29)
	at io.cucumber.core.runner.CoreStepDefinition.execute(CoreStepDefinition.java:66)
	at io.cucumber.core.runner.PickleStepDefinitionMatch.runStep(PickleStepDefinitionMatch.java:63)
	at io.cucumber.core.runner.ExecutionMode$1.execute(ExecutionMode.java:10)
	at io.cucumber.core.runner.TestStep.executeStep(TestStep.java:92)
	at io.cucumber.core.runner.TestStep.run(TestStep.java:64)
	at io.cucumber.core.runner.PickleStepTestStep.run(PickleStepTestStep.java:51)
	at io.cucumber.core.runner.TestCase.run(TestCase.java:104)
	at io.cucumber.core.runner.Runner.runPickle(Runner.java:73)
	at io.cucumber.junit.PickleRunners$NoStepDescriptions.run(PickleRunners.java:151)
	at io.cucumber.junit.FeatureRunner.runChild(FeatureRunner.java:135)
	at io.cucumber.junit.FeatureRunner.runChild(FeatureRunner.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at io.cucumber.junit.CucumberSerenityRunner.runChild(CucumberSerenityRunner.java:235)
	at io.cucumber.junit.CucumberSerenityRunner.runChild(CucumberSerenityRunner.java:54)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at io.cucumber.junit.CucumberSerenityRunner$RunCucumber.evaluate(CucumberSerenityRunner.java:263)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:221)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)

**Comment by wakaleo** [comment]
"Unsupported class file major version 60" - that looks like some issue related to JDK versions. What version of Java are you using?

**Comment by TarahMo** [comment]
I am on java version "16.0.1"

**Comment by wakaleo** [comment]
Try with 15

**Comment by TarahMo** [comment]
Sorry for the late response - I was exploring a few things in customizing the serenity tests for the 'Best Buy API Playground' API hosted on my localhost to check if they go through with the set-up. Setting up JDK 15 did help and it resolved the issues highlighted in my previous post. I updated the JDK version under the project structure option inside Intellij.

For the 'Best Buy API Playground' API hosted on my localhost, I am able to fire multiple GET & POST requests and access the beautiful serenity reports. As a matter of fact, I am running the project using Maven Run configuration by setting up goal = clean verify serenity:aggregate. ALL good.

thank you so much for your time John. I must mention your responses were incredibly quick which helped me move faster in resolving the errors I encountered, really appreciate and thanks a ton again.
